modules/storage-api/src/main/java/org/apache/ignite/internal/storage/util/StorageUtils.java

```diff
@@ -137,6 +138,7 @@ public static void throwExceptionDependingOnStorageStateOnRebalance(StorageState
     public static void throwExceptionDependingOnStorageState(StorageState state, String storageInfo) {
         switch (state) {
             case CLOSED:
+                assert !IgniteSystemProperties.enabledColocation() : createStorageClosedErrorMessage(storageInfo);
```

Comment:
Why is this line needed?

Reply:
I want to be sure that while colocation track we'll see every preventive storage closes. I add TODO for colocation cleanup

Reply:
Because the already stopped exception just will be shown in logs silently and all tests will be fine. With the assertion we will have a failed TC suite.

Reply:
Do you have a precedent for this? Because it looks very strange now. For example, why do you only check the CLOSED status and no other statuses? Why does accessing a closed storage not result in test failures?

Reply:
>Do you have a precedent for this?

E.g. I ran `testTableEstimatedSize` as `@RepeatedTest(20)` and 15 out of 20 contain `Storage is already closed`.

>Why does accessing a closed storage not result in test failures?

In most of runs previously there was the exception on a node stop, but it didn't affect on TC runs, because it happend on `@AfterEach` without fail behavior. Our TC reacts only for `AssertionError` and `NullPointerException` thus we couldn't detect the closed exception before.

>why do you only check the CLOSED status and no other statuses?

Because this state disturbs me most.

Reply:
>why would we suddenly stop being interested in storages being stopped incorrectly?

Because after colocation we will cover all scenarios and we'll be sure with them, but

>I think what you want to do is to remove this line

Yes, done.

>fix TC settings to fail if this exception is detected in the logs

It should be a request to `it-assistance`?

Reply:
> It should be a request to it-assistance?

I don't know, but let's try